### PR TITLE
♻️Pydantic V2 migration: various fixes

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

- fixed typing issue in aws-library -> green
- fixed weird assert in resource-usage-tracker + return proper AnyUrl --> green
- fixed wrong requirement in dynamic-scheduler
- fixed wrong imports here and there

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Review by matusdrobuliak66 — APPROVED** [review]
:+1:

**Review by giancarloromeo — APPROVED** [review]
Yup!

File: packages/simcore-sdk/src/simcore_sdk/node_ports_v2/nodeports_v2.py [hunk 1]
```diff
@@ -28,6 +30,28 @@
 log = logging.getLogger(__name__)
 
 
+#  -> @GitHK this looks very dangerous, using a lot of protected stuff, just checking the number of ignores shows it's a bad idea...
```
Comment (GitHK) [inline packages/simcore-sdk/src/simcore_sdk/node_ports_v2/nodeports_v2.py:33]: Maybe a test could be added so is caught in case something goes wrong?

**Review by GitHK — APPROVED** [review]

**Review by pcrespov — APPROVED** [review]
Thx a lot!. From now on, i suggest that we do the sync with our own commits to master. This way you you do not have to be doing this by your own :-)

File: packages/simcore-sdk/src/simcore_sdk/node_ports_v2/nodeports_v2.py (hunk 1 above)
Comment (sanderegg) [reply packages/simcore-sdk/src/simcore_sdk/node_ports_v2/nodeports_v2.py:33]: Please be my guest. that is your code @GitHK and these 2 functions were added by you. I did not try to understand too much of them, they look complex enough.

**Comment by sanderegg** [comment]
> Thx a lot!. From now on, i suggest that we do the sync with our own commits to master. This way you you do not have to be doing this by your own :-)

@crespov: not sure I understand what you mean here

**Pull request merged**